planmanager.py

```diff
@@ -26,19 +28,24 @@ def get_plans() -> List[SPlans]:
         This is to ensure that the plans are not corrupted after the
         lock is released.
         """
-        PlanManager._pm_lock.acquire()
-        plans = deepcopy(PlanManager._plans)
-        PlanManager._pm_lock.release()
+        # PlanManager._pm_lock.acquire()
+        with shelve.open('plans', flag='r') as db: 
+            plans = deepcopy(db['plans'])
+        #PlanManager._pm_lock.release()
         return plans
 
     @staticmethod
     def set_plans(plans: List[Plans]) -> NoReturn:
         """
         Note that we are converting List[Plans] to List[SPlans].
         """
-        PlanManager._pm_lock.acquire()
+        # PlanManager._pm_lock.acquire()
```

Comment:
Why are we getting rid of the lock? Even the `shelve` page says that we have to manage locks ourselves, and that we should always use a lock when writing to prevent any other writing or reading from being done.